Components.md

```diff
@@ -0,0 +1,137 @@
+# Core
+>Spark Core is the base of the whole project. It provides distributed task dispatching, scheduling, and basic I/O functionalities. Spark uses a specialized fundamental data structure known as RDD (Resilient Distributed Datasets) that is a logical collection of data partitioned across machines. RDDs can be created in two ways; one is by referencing datasets in external storage systems and second is by applying transformations (e.g. map, filter, reducer, join) on existing RDDs.
+
+## Responsible for
+* Memory managment
+* Fault recovery
+* Scheduling, distributing and monitoring jobs on a cluster
+* Interacting with storage system
+
+## RDD (Resilient Distributed Dataset)
+> Spark Core is embedded with RDDs, an immutable fault-tolerant, destributed collection of objects that can be operated on in parallel.
+> <br>Working with RDD you need to know 2 main things: transformations and actions.
+* RDD transformations returns pointer to new RDD and allows you to create dependencies between RDDs. Each RDD in dependency chain (String of Dependencies) has a function for calculating its data and has a pointer (dependency) to its parent RDD. Spark is lazy, so nothing will be executed unless you call some transformation or action that will trigger job creation and execution. Look at the following snippet of the word-count example. Therefore, RDD transformation is not a set of data but is a step in a program telling Spark how to get data and what to do with it.
+* RDD actions are operations(reduce, count...) that return the value according to its instruction 
+![Markdown Logo](img/rdd.png)
+## Example (Creating RDD)
```

Comment:
I suppose, it would be fine to add some links or manual for installation Spark locally, what do you think?
It allows to try some commands from code below

Reply:
Yes, I added guide for installation